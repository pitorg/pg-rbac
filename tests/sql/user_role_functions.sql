SELECT '63d412c1-6752-42f2-a71b-be7f8dda83da'::UUID AS user_id \gset
SELECT rbac.add_user(:'user_id'::UUID);
SELECT rbac.add_role('cashier') AS cashier_role_id \gset
SELECT rbac.add_role('manager') AS manager_role_id \gset
SELECT rbac.add_role('admin') AS admin_role_id \gset
SELECT rbac.assign_user(:'user_id'::UUID, :'cashier_role_id'::UUID);
SELECT rbac.assign_user(:'user_id'::UUID, :'admin_role_id'::UUID);

SELECT roles.name
FROM rbac.assigned_users
JOIN rbac.roles ON roles.id = assigned_users.role_id
WHERE assigned_users.user_id = :'user_id'::UUID
ORDER BY roles.name;

\set VERBOSITY terse
SELECT rbac.assign_user(:'user_id'::UUID, :'admin_role_id'::UUID); -- error since already assigned
SELECT rbac.assign_user(:'user_id'::UUID, 'cdca237a-a240-448e-a4c5-7ae747d0156e'::UUID); -- error since non-existing role
SELECT rbac.assign_user('cdca237a-a240-448e-a4c5-7ae747d0156e'::UUID, :'admin_role_id'::UUID); -- error since non-existing user
