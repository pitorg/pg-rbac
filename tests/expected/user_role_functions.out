SELECT '63d412c1-6752-42f2-a71b-be7f8dda83da'::UUID AS user_id \gset
SELECT rbac.add_user(:'user_id'::UUID);
 add_user 
----------
 t
(1 row)

SELECT rbac.add_role('cashier') AS cashier_role_id \gset
SELECT rbac.add_role('manager') AS manager_role_id \gset
SELECT rbac.add_role('admin') AS admin_role_id \gset
SELECT rbac.assign_user(:'user_id'::UUID, :'cashier_role_id'::UUID);
 assign_user 
-------------
 t
(1 row)

SELECT rbac.assign_user(:'user_id'::UUID, :'admin_role_id'::UUID);
 assign_user 
-------------
 t
(1 row)

SELECT roles.name
FROM rbac.assigned_users
JOIN rbac.roles ON roles.id = assigned_users.role_id
WHERE assigned_users.user_id = :'user_id'::UUID
ORDER BY roles.name;
  name   
---------
 admin
 cashier
(2 rows)

\set VERBOSITY terse
SELECT rbac.assign_user(:'user_id'::UUID, :'admin_role_id'::UUID); -- error since already assigned
ERROR:  duplicate key value violates unique constraint "assigned_users_pkey"
SELECT rbac.assign_user(:'user_id'::UUID, 'cdca237a-a240-448e-a4c5-7ae747d0156e'::UUID); -- error since non-existing role
ERROR:  insert or update on table "assigned_users" violates foreign key constraint "assigned_users_role_id_fkey"
SELECT rbac.assign_user('cdca237a-a240-448e-a4c5-7ae747d0156e'::UUID, :'admin_role_id'::UUID); -- error since non-existing user
ERROR:  insert or update on table "assigned_users" violates foreign key constraint "assigned_users_user_id_fkey"
